index.js

```diff
@@ -78,24 +78,48 @@ const DEFAULT_CONFIG_OPTIONS = {
     health: true,
 };
 
+function routesConfig(config, defaultConfig) {
+    if (config === true) {
+        return defaultConfig;
+    } else if (config === false) {
+        return {
+            ...defaultConfig,
+            timing: false,
+            hits: false,
+            errors: false,
+        };
+    } else {
+        return {
+            ...defaultConfig,
+            ...config,
+        };
+    }
+}
+
 module.exports = fp(
     async function (fastify, opts) {
         const { client = {}, routes, ...others } = opts;
-        // TODO: allow to pass routes: false to disable the routes alltoghether.
+        if (opts.routes) {
```

Comment:
Why don't you use only one `if`? as:
```js
if (opts.routes &&
    typeof opts.routes !== 'boolean' &&
    typeof opts.routes !== 'object') {
          throw new Error('"routes" must be a boolean or a config object.');
}
```

Reply:
I though it would have been simpler to read, I'll change it 😉

Reply:
No problem, if you think it is more readable you can maintain it. I have the opposite opinion, more indentations there are more unreadable is.